actors/eam/src/lib.rs

```diff
@@ -0,0 +1,289 @@
+use std::iter;
+
+use ext::init::{Exec4Params, Exec4Return};
+use rlp::Encodable;
+
+mod ext;
+
+use {
+    fil_actors_runtime::{
+        actor_error, cbor,
+        runtime::builtins::Type,
+        runtime::{ActorCode, Runtime},
+        ActorError, EAM_ACTOR_ID, INIT_ACTOR_ADDR,
+    },
+    fvm_ipld_blockstore::Blockstore,
+    fvm_ipld_encoding::{strict_bytes, tuple::*, RawBytes},
+    fvm_shared::{
+        address::{Address, Payload, SECP_PUB_LEN},
+        crypto::hash::SupportedHashes,
+        ActorID, MethodNum, METHOD_CONSTRUCTOR,
+    },
+    num_derive::FromPrimitive,
+    num_traits::FromPrimitive,
+};
+
+#[cfg(feature = "fil-actor")]
+fil_actors_runtime::wasm_trampoline!(EamActor);
+
+/// Maximum allowed EVM bytecode size.
+/// The contract code size limit is 24kB.
+const MAX_CODE_SIZE: usize = 24 << 10;
+
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    Create = 2,
+    Create2 = 3,
+    // CreateAccount = 4,
+}
+
+#[derive(Debug)]
+/// Intermediate type between RLP encoding for CREATE
+struct RlpCreateAddress {
+    address: EthAddress,
+    nonce: u64,
+}
+
+impl rlp::Encodable for RlpCreateAddress {
+    fn rlp_append(&self, s: &mut rlp::RlpStream) {
+        s.encoder().encode_value(&self.address.0);
+        s.append(&self.nonce);
+    }
+}
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone, Copy)]
+pub struct EthAddress(#[serde(with = "strict_bytes")] pub [u8; 20]);
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct CreateParams {
+    #[serde(with = "strict_bytes")]
+    pub initcode: Vec<u8>,
+    pub nonce: u64,
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct Create2Params {
+    #[serde(with = "strict_bytes")]
+    pub initcode: Vec<u8>,
+    /// TODO are we hashing with Little Endian bytes
```

Comment:
Where? How? Can we fix it?

Reply:
comment is a lie. we dont change any endianness from the input.